src/main/java/school/hei/haapi/endpoint/rest/mapper/FileInfoMapper.java

```diff
@@ -24,4 +28,20 @@ public school.hei.haapi.endpoint.rest.model.FileInfo toRest(FileInfo fileInfo) {
         .fileUrl(presignedUrl)
         .creationDatetime(fileInfo.getCreationDatetime());
   }
+
+  public ShareInfo toShareInfo(OcsData ocsData) {
+    log.info("sssssss");
+    return new ShareInfo()
+        .name(ocsData.getName())
+        .path(ocsData.getPath())
+        .url(ocsData.getUrl())
+        .permissions(
+            switch (ocsData.getPermissions()) {
+              case 1 -> "READ";
+              case 2 -> "READ/WRITE";
```

Comment:
hah finally some use of my Effective Java Knowledge.

create an enum named OwnCloudFilePermissions, with the needed values, then return an EnumSet in the permissions attribute